src/schema/slack/functions/_scripts/src/templates/template_function.ts

```diff
@@ -0,0 +1,149 @@
+import { isCustomType } from "../../../../../../types/mod.ts";
+import SchemaTypes from "../../../../../schema_types.ts";
+import SlackSchemaTypes from "../../../../schema_types.ts";
+import { InternalSlackTypes } from "../../../../types/custom/mod.ts";
+import {
+  FunctionParameter,
+  FunctionProperties,
+  FunctionProperty,
+  FunctionRecord,
+} from "../types.ts";
+import {
+  autogeneratedComment,
+  getSlackCallbackId,
+  renderTypeImports,
+} from "./template_utils.ts";
+import { AllowedTypeValue, AllowedTypeValueObject } from "./types.ts";
+
+const typeMap: Record<string, AllowedTypeValueObject> = {
+  SchemaTypes,
+  SlackTypes: SlackSchemaTypes,
+  InternalSlackTypes,
+};
+
+const schemaTypeMap = Object.entries(typeMap).reduce<Record<string, string>>(
+  (acc, [schemaKey, schemaTypes]) => {
+    for (const schemaType in schemaTypes) {
+      const value: AllowedTypeValue = schemaTypes[schemaType];
+      const type: string = isCustomType(value) ? value.id : value;
+      acc[type] = `${schemaKey}.${schemaType}`;
+    }
+    return acc;
+  },
+  {},
+);
+
+const propertyToTypeScript = (
+  property: FunctionProperty,
+): string => {
+  const typescript = [];
+  typescript.push(
+    `type: ${schemaTypeMap[property.type]}`,
+  );
+  if (property.description) {
+    typescript.push(`description: "${property.description}"`);
+  }
+  if (property.title) {
+    typescript.push(`title: "${property.title}"`);
+  }
+  if ("items" in property) {
+    typescript.push(`items: ${propertyToTypeScript(property.items)}`);
+  }
+  if ("properties" in property) {
+    typescript.push(`additionalProperties: ${property.additionalProperties}`);
```

Comment:
The property in this case is a `FunctionProperty` that is an object, so because it comes from how the backend stores functions there may be a shape difference between what we require — tbh I don't know if the backend fills `additionalProperties` and `required` when it stores them. Safe bet is to handle the undefined case!

Reply:
Good catch!!
I was also unsure about what was required and what wasn't, will fix to deal with worst case